src/ore/src/process.rs

```diff
@@ -45,3 +45,73 @@ macro_rules! halt {
             ::std::process::exit(2);
     }}
 }
+
+#[cfg(feature = "metrics")]
+pub mod rusage {
```

Comment:
It feels weird to me to have this in ore. What do you think about spinning up a `mz-metrics` crate? I've been meaning to move all the metrics stuff to a dedicated crate, and this would be a nice way to kick it off.

Reply:
More generally, is there an advantage of having a big `mz-ore` crate with features vs. many small ones without features? It seems like the latter approach would be beneficial for compile times because fewer things need to be recompiled if you only change code related to one of the features currently under the `mz-ore` umbrella.

Reply:
Extracted into a new `mz-metrics` crate, only containing the rusage metrics at this point.